filtering/checker.go

```diff
@@ -155,6 +157,11 @@ func (c *Checker) resolveCallExprFunctionOverload(
 				if !ok {
 					return nil, c.errorf(callExpr.Args[i], "unknown type")
 				}
+
+				if argType.GetPrimitive() == expr.Type_UINT64 && param.GetPrimitive() == expr.Type_INT64 {
+					param = TypeUint
```

Comment:
This looks hacky, why not register a proper overload?

Reply:
@srstack We've discussed this PR and our feedback is that adding an if statement here seems like a workaround that increases the complexity of the implementation and makes it harder to understand.

Have you considered solving your use case of `foo = 10 OR foo = 10000000000000000001` by registering a `uint` function that will let you say `foo = uint(10) OR foo = 10000000000000000001`?

This seems like the idiomatic Go solution (be explicit rather than implicit).

Reply:
If you remove this if statement we are OK to merge the remainder of the PR!

Reply:
LGTM ~